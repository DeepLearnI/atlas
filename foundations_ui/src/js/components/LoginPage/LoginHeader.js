import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LoginHeader extends Component {
  constructor(props) {
    super(props);
    this.state = {
      pageTitle: this.props.pageTitle,
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ pageTitle: nextProps.pageTitle });
  }

  render() {
    const { pageTitle } = this.state;

    return (
      <div className="project-header-container">
        <div className="project-header-logo-container">
          <div className="i--icon-logo" />
          <h2 className="font-bold">Foundations</h2>
        </div>
        <div className="project-header-info-container">
          <div className="project-header-total-projects-container">
            <div className="half-width inline-block">
              <h1 className="blue-border-bottom font-bold">{ pageTitle }</h1>
            </div>
          </div>
          <div className="project-header-sort-filter-container" />
        </div>
      </div>
    );
  }
}

LoginHeader.propTypes = {
  pageTitle: PropTypes.string,
};

LoginHeader.defaultProps = {
  pageTitle: '',
};

export default LoginHeader;
