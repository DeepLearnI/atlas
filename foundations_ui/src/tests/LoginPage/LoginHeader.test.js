import React from 'react';
import LoginHeader from '../../js/components/LoginPage/LoginHeader';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders LoginHeader', () => {
  const wrapper = shallow(<LoginHeader/>);
  expect(wrapper).toMatchSnapshot();
});