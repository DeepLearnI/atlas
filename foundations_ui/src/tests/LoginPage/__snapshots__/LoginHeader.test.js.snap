// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders LoginHeader 1`] = `
<div
  className="project-header-container"
>
  <div
    className="project-header-logo-container"
  >
    <div
      className="i--icon-logo"
    />
    <h2
      className="font-bold"
    >
      Foundations
    </h2>
  </div>
  <div
    className="project-header-info-container"
  >
    <div
      className="project-header-total-projects-container"
    >
      <div
        className="half-width inline-block"
      >
        <h1
          className="blue-border-bottom font-bold"
        />
      </div>
    </div>
    <div
      className="project-header-sort-filter-container"
    />
  </div>
</div>
`;
